### PR TITLE
Added Promise.any polyfill + tests

### PR DESCRIPTION
**Pull request by spanic**
Hi guys, I implemented missing [Promise.any function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/any), could you please check?

Tested locally with node v0.10.48 (npm v2.15.1) - works as expected. Also I added some unit tests to cover new functionality - all of them are passed.

_P. S. Feel free to ask me any questions: this is my first PR, so maybe I missed something important. Thank you!_

### CODE REVIEW

File: dist/polyfill.js [hunk 1]
```diff
@@ -277,6 +286,33 @@ Promise.all = function(arr) {
   });
 };
 
+Promise.any = function(arr) {
```
Comment (taylorhakes) [inline dist/polyfill.js:289]: Let's add this code to both index.js and polyfill.js. Might make sense to abstract this function into a separate file

**Comment by taylorhakes** [comment]
@spanic This looks great. Really appreciate your work on this. One minor comment about updating the polyfill inside src

File: dist/polyfill.js (hunk 1 above)
Comment (spanic) [reply dist/polyfill.js:289]: Sure, let me try, I'll update this request when it will be done 👌🏻

**Comment by spanic** [comment]
@taylorhakes could you please check the latest update? 🤔

**Pull request merged**